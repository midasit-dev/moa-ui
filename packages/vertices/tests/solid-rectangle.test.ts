import { toVertices } from "@lib-functions";
import { Vertex2D } from '@lib-types';

describe('functions/solid-rectangle.ts', () => {
	test('250x300', () => {
		const vertices: Vertex2D[][] | Error = toVertices({
			type: "SolidRectangle",
			properties: {
				b: 250,
				h: 300,
			},
		});

		if (vertices instanceof Error) {
			fail(`Function returned an error: ${vertices.message}`);
		} else {
			expect(vertices).toMatchSnapshot();
		}
	});
});