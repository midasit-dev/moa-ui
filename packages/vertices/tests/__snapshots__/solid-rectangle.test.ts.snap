// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions/solid-rectangle.ts 250x300 1`] = `
[
  [
    {
      "x": -125,
      "y": 150,
    },
    {
      "x": 125,
      "y": 150,
    },
    {
      "x": 125,
      "y": -150,
    },
    {
      "x": -125,
      "y": -150,
    },
  ],
]
`;
